### PR TITLE
Run lintian inside the build environement

### PR DESCRIPTION
**Pull request by hashar**
scripts/lintian-junit-report first runs lintian on the build host and
then post process the output to generate a JUnit file.

I have the use case to run lintian via a pbuilder hook to use the
lintian version that comes with the targetted distribution. The result
is a lintian.txt which I need to convert to JUnit.

Merge in e2beae591b from the lintian branch, that lets one skip running lintian
and instead process an existing lintian output.

Add a new pbuilder hook B90lintian based on the autopkgtest. By setting the
environment variable LINTIAN=true, the hook will: 
* install lintian
* run it against the package

pbuilder MUST have ADDITIONAL_BUILDRESULTS=(../*.lintian.txt). That would
export the lintian result out of the build environement to pbuilder buildresult
directory.  One can then process the file using something like:

for x in *.lintian.txt do; 
    echo "Converting lintian file $x" 
    /usr/bin/lintian-junit-report \
      --skip-lintian \
      --filename "$x" \
      *.changes > "${x/$suffix/.lintian.xml}"
done

### CODE REVIEW

**Comment by hashar** [comment]
It seems to work for me. The Jenkins job integration I have made is a work in progress and can be seen at https://gerrit.wikimedia.org/r/#/c/408268/

I will try to update the JJB recipe you have.

**Comment by hashar** [comment]
e2beae5  by @linuxmaniac and my hook d59f111 have been working fine so far. Someone might want to give it a second look, but I think it is safe to merge :]

File: pbuilder-hookdir/B90lintian [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+if [ -n "${LINTIAN_OPTIONS:-}" ] ; then
+  echo "*** Using provided LINTIAN_OPTIONS $LINTIAN_OPTIONS ***"
+fi
+
+set -ex -o pipefail
+apt-get -y "${APTGETOPT[@]}" install lintian
+lintian --version
+
+PACKAGE_NAME=$(basename "$(realpath "$BUILDDIR"/*/debian/..)")
```
Comment (hashar) [inline pbuilder-hookdir/B90lintian]: Sometime `realpath` will not be available. It is provided by `coreutils`:

 $ dpkg -S /usr/bin/realpath
 coreutils: /usr/bin/realpath

There might be a way to extract the package name without `realpath` though. Else the hook has to install the `coreutils` package as well.

File: pbuilder-hookdir/B90lintian [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+if [ -n "${LINTIAN_OPTIONS:-}" ] ; then
+  echo "*** Using provided LINTIAN_OPTIONS $LINTIAN_OPTIONS ***"
+fi
+
+set -ex -o pipefail
+apt-get -y "${APTGETOPT[@]}" install lintian
+lintian --version
+
+package_dir=$(cd "$BUILDDIR"/*/debian/.. 2>/dev/null && pwd -P)
+PACKAGE_NAME=$(basename "$package_dir")
```
Comment (hashar) [inline pbuilder-hookdir/B90lintian]: Updated to no more rely on realpath. Deployed on Wikimedia infrastructure and it works! https://phabricator.wikimedia.org/T193906

**Comment by hashar** [comment]
@mika I had those commits in my version of jenkins-debian-glue and they are working properly :)

Wikimedia downstream bug is https://phabricator.wikimedia.org/T186494

**Comment by mika** [comment]
Thanks a lot, @hashar - trying to get this into Debian/buster, LGTM + just merged it! \o/

**Comment by hashar** [comment]
Thanks! I am looking forward to install v0.20.0 :-)

**Pull request closed without merging**